docs/preview/02-Guidance/Service-to-service Correlation/use-service-to-service-correlation-in-web-api.md

```diff
@@ -0,0 +1,312 @@
+
+### Product API: startup code
+First, lets look the initial code for the **Product API**. The startup code has the hosting and routing functionality to use API controllers. The configuration value `STOCK_API_URL` already reveals that this API will contact another API. This is also the reason why we added `.AddHttpClient()` to the startup code so we can create `HttpClient` instances. ([More information on creating HTTP clients in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-6.0)). 
+```csharp
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        WebApplicationBuilder builder = WebApplication.CreateBuilder(args);
+        builder.Services.AddRouting();
+        builder.Services.AddControllers();
+        
+        builder.Configuration.AddInMemoryCollection(new[]
+        {
+            new KeyValuePair<string, string>("STOCK_API_URL", "http://localhost:788")
+        });
+        builder.Services.AddHttpClient("Stock API");
+
+        WebApplication app = builder.Build();
+        app.UseRouting();
+        app.UseEndpoints(endpoints => endpoints.MapControllers());
+        app.Run("http://localhost:787");
+    }
+}
+```
+
+The sole API controller in the **Product API** makes sure that we receive the product order request, contact the **Stock API** to request the available stock for that product, and determine the user response based on that availability.
+```csharp
+using System.Net.Http;
+using System.Net.Http.Json;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+
+public class ProductOrderRequest
+{
+    public string ProductName { get; set; }
+    public int Amount { get; set; }
+}
+
+[ApiController]
+[Route("api/v1/product")]
+public class ProductController : ControllerBase
+{
+    private readonly IHttpClientFactory _httpClientFactory;
+    private readonly IConfiguration _configuration;
+
+    public ProductController(
+        IHttpClientFactory httpClientFactory, 
+        IConfiguration configuration)
+    {
+        _httpClientFactory = httpClientFactory;
+        _configuration = configuration;
+    }
+
+    [HttpPost]
+    public async Task<IActionResult> Post([FromBody] ProductOrderRequest request)
+    {
+        HttpClient httpClient = _httpClientFactory.CreateClient("Stock API");
+        
+        var endpoint = $"{_configuration["STOCK_API_URL"]}/api/v1/stock?productName={orderRequest.ProductName}";
+
+        var response = await httpClient.GetFromJsonAsync<ProductStockResponse>(endpoint);
+        if (response.AvailableItems >= orderRequest.Amount)
+        {
+            return Accepted();
+        }
+
+        return StatusCode(StatusCodes.Status406NotAcceptable);
```

Comment:
I don't think this is the right status code to return, according to Wikipedia (https://en.wikipedia.org/wiki/List_of_HTTP_status_codes): `The requested resource is capable of generating only content not acceptable according to the Accept headers sent in the request.`

I would use HTTP 500 for this, as it isn't really a client error (4xx). Or use HTTP 418 of course 🫖😉

Reply:
> Or use HTTP 418 of course 🫖😉

Hehe

Reply:
Done in https://github.com/arcus-azure/arcus.observability/pull/446/commits/8d548d4012214107b4f3742f2fe436d3d05519e2